File: 1._File_formats_bonanza/node/read_and_parse.js
// Written: Roos Lindeboom 30 aug 2022


// --- CSV ---
const fs = require('fs');
 
fs.readFile('/Users/rooslindeboom/Desktop/KEA - projects/System integrations/SI-homework/systemIntegrations_homework/1._File_formats_bonanza/node/Article_files/Article.cvs', 'utf8', function (err, data) {
  console.log("-- cvs --");  
  console.log(data)
});

// --- yaml ---
const yaml = require('js-yaml');

try {
    let fileContents = fs.readFileSync('/Users/rooslindeboom/Desktop/KEA - projects/System integrations/SI-homework/systemIntegrations_homework/1._File_formats_bonanza/node/Article_files/Article.yaml', 'utf8');
    let data = yaml.load(fileContents);

    console.log("-- yaml --");
    console.log(data);
} catch (e) {
    console.log(e);
}

File: 1._File_formats_bonanza/python/read_and_parse.py
# Written: Roos Lindeboom 30 aug 2022

import json

# --- TXT ---
# Read file text
print("-- TXT --\n")
with open('/Users/rooslindeboom/Desktop/KEA - projects/System integrations/SI-homework/systemIntegrations_homework/1._File_formats_bonanza/python/Article_files/Article.txt') as f:
    contents = f.read()
    print(contents)


# --- XML ---
print("\n-- XML --\n")

# Read file XML
from xml.dom import minidom

# parse an xml file by name
file = minidom.parse('/Users/rooslindeboom/Desktop/KEA - projects/System integrations/SI-homework/systemIntegrations_homework/1._File_formats_bonanza/python/Article_files/Article.xml')

#use getElementsByTagName() to get tag
article = file.getElementsByTagName('article')


for elem in article:
  print(elem.firstChild.data)


# --- JSON ---
print("\n-- JSON --\n")
# Opening JSON file
f = open('/Users/rooslindeboom/Desktop/KEA - projects/System integrations/SI-homework/systemIntegrations_homework/1._File_formats_bonanza/python/Article_files/Article.json')
  
# returns JSON object as 
# a dictionary
data = json.load(f)
  
# Iterating through the json
# list
print(data['article'])
  
# Closing file
f.close()